src/poke_env/player/player.py

```diff
@@ -702,17 +706,30 @@ async def battle_against(self, opponent: "Player", n_battles: int = 1):
         :param n_battles: The number of games to play. Defaults to 1.
         :type n_battles: int
         """
-        await handle_threaded_coroutines(self._battle_against(opponent, n_battles))
-
-    async def _battle_against(self, opponent: "Player", n_battles: int):
-        await asyncio.gather(
-            self.send_challenges(
-                to_id_str(opponent.username),
-                n_battles,
-                to_wait=opponent.ps_client.logged_in,
-            ),
-            opponent.accept_challenges(to_id_str(self.username), n_battles),
+        result = await handle_threaded_coroutines(
+            self._battle_against(*opponents, n_battles=n_battles)
         )
+        return result
+
+    async def _battle_against(
+        self, *opponents: Player, n_battles: int
+    ) -> Dict[str, Tuple[float, float]]:
+        results: Dict[str, Tuple[float, float]] = {}
+        for opponent in opponents:
+            await asyncio.gather(
+                self.send_challenges(
+                    to_id_str(opponent.username),
+                    n_battles,
+                    to_wait=opponent.ps_client.logged_in,
+                ),
+                opponent.accept_challenges(
+                    to_id_str(self.username), n_battles, opponent.next_team
+                ),
+            )
+            results[opponent.username] = (self.win_rate, opponent.win_rate)
```

Comment:
This should bot be part of this method - reporting results is unrelated to starting battles. Similarly, resetting the stored battles after the battles are done would be counterintuitive and make for a bad API.

Reply:
Good point! Here you go: f17c415